p5/sketch/userspace.py

```diff
@@ -143,34 +139,35 @@ def run(sketch_setup=None, sketch_draw=None,
             hfunc = getattr(__main__, handler)
             handlers[handler] = _fix_interface(hfunc)
 
-    if mode == "P2D":
-        p5.mode = 'P2D'
-        if renderer == "vispy":
-            p5.renderer = Renderer2D()
-        else:
-            raise NotImplementedError("Invalid Renderer %s" % renderer)
-    elif mode == "P3D":
-        p5.mode = 'P3D'
-        if renderer == "vispy":
+    if renderer == "vispy":
+        import vispy
+        vispy.use('glfw')
+        from p5.sketch.Vispy2DRenderer.base import VispySketch
+        from vispy import app
+        builtins.current_renderer = "vispy"
+        if mode == "P2D":
+            from p5.sketch.Vispy2DRenderer.renderer2d import VispyRenderer2D
+            p5.mode = 'P2D'
+            p5.renderer = VispyRenderer2D()
+        elif mode == "P3D":
+            p5.mode = 'P3D'
             p5.renderer = Renderer3D()
         else:
-            raise NotImplementedError("Invalid Renderer %s" % renderer)
-    else:
-        raise ValueError("Invalid Mode %s" % mode)
-
-    p5.sketch = Sketch(setup_method, draw_method, handlers, frame_rate)
+            ValueError("Invalid Mode %s" % mode)
+        p5.sketch = VispySketch(setup_method, draw_method, handlers, frame_rate)
+        physical_width, physical_height = p5.sketch.physical_size
+        width, height = p5.sketch.size
 
-    physical_width, physical_height = p5.sketch.physical_size
-    width, height = p5.sketch.size
+        builtins.pixel_x_density = physical_width // width
+        builtins.pixel_y_density = physical_height // height
+        builtins.start_time = time.perf_counter()
 
-    builtins.pixel_x_density = physical_width // width
-    builtins.pixel_y_density = physical_height // height
-    builtins.start_time = time.perf_counter()
+        p5.sketch.timer.start()
 
-    p5.sketch.timer.start()
-
-    app.run()
-    exit()
+        app.run()
```

Comment:
This line is visy specific. I believe if we use another renderer, it will throw an error.

Reply:
`app.run()` will be executed only when the renderer is set to `vispy`. 
`app.run()` and all other `vispy` specific calls is under the `if renderer == "vispy"` clause. I think Github's UI is making it confusing to figure out the indents.

Or is it something else that I am missing, can you confirm this @parsoyaarihant ?
Thank you for the review :)